### PR TITLE
Support new IDE states, parsing and preparing verification

### PR DESCRIPTION
**Pull request by keyboardDrummer**
### Testing

- New statuses are shown when using Dafny master
- Statuses still work as they did when using Dafny 4.0

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -287,9 +287,6 @@
       }
     ]
   },
-  "activationEvents": [
```
Comment (keyboardDrummer) [inline package.json:290]: This is inferred from the language being added in the package.json

File: src/ui/compilationStatusView.ts [hunk 2]
```diff
@@ -36,6 +40,7 @@ function toStatusMessage(status: CompilationStatus, message?: string | null): st
       ? `${Messages.CompilationStatus.VerificationFailedOld} ${message}`
       : `${Messages.CompilationStatus.VerificationFailedOld}`;
   }
+  return status.toString();
```
Comment (MikaelMayer) [inline src/ui/compilationStatusView.ts:43]: This is for future forward compatibility, rivht?

**Review by MikaelMayer — APPROVED** [review]
Thanks for the follow up !

File: src/ui/compilationStatusView.ts (hunk 2 above)
Comment (keyboardDrummer) [reply src/ui/compilationStatusView.ts:43]: I don't think it's needed for the current changes, but I thought it might be good defensive programming

**Pull request merged**